app/controllers/metrics_controller.rb

```diff
@@ -1,19 +1,26 @@
 class MetricsController < ApplicationController
+  include Concerns::DateRangeParser
+
   DEFAULT_METRICS = %w[pageviews unique_pageviews number_of_internal_searches].freeze
 
   def show
     service = MetricsService.new
 
+    date_range = parse_date_range(params[:date_range])
+    from = date_range[:from]
+    to = date_range[:to]
+
     service_params = {
       base_path: params[:base_path],
-      from: params[:from],
-      to: params[:to],
+      from: from,
+      to: to,
```

Comment:
If we extract a DateRange, we can update the service parameters?

Reply:
The `date_range` has `to` and `from` attributes. Not sure why we would need to change it?